internal/worker/dbaccessor/worker_test.go

```diff
@@ -505,85 +499,38 @@ func (s *workerSuite) TestWorkerStartupAsBootstrapNodeThenReconfigureWithLoopbac
 
 	// Ensure that we expect a clean startup and shutdown.
 	s.expectNodeStartupAndShutdown()
-	s.expectNoConfigChanges()
 
-	s.hub.EXPECT().Subscribe(apiserver.DetailsTopic, gomock.Any()).Return(func() {}, nil)
-
-	s.client.EXPECT().Cluster(gomock.Any()).Return(nil, nil)
+	// First time though, there is no config, then we get a
+	// notification for a change on disk.
+	ch := make(chan struct{})
+	s.expectConfigChanges(ch)
 
-	w := s.newWorker(c)
-	defer func() {
-		close(dbDone)
-		// We do want a clean kill here, because we're not rebinding.
-		workertest.CleanKill(c, w)
-	}()
-	dbw := w.(*dbWorker)
-
-	ensureStartup(c, dbw)
-
-	// At this point we have started successfully.
-	// Push a message onto the API details channel to simulate changes.
-	select {
-	case dbw.apiServerChanges <- apiserver.Details{
-		Servers: map[string]apiserver.APIServer{
-			"0": {ID: "0", InternalAddress: "127.0.0.1:1234"},
-		},
-	}:
-	case <-time.After(testing.LongWait):
-		c.Fatal("timed out waiting for cluster change to be processed")
-	}
-}
-
-func (s *workerSuite) TestWorkerStartupAsBootstrapNodeThenReconfigureWithLoopbackPreferredAndNotLoopbackBound(c *gc.C) {
```

Comment:
This test was removed rather than being rewritten because it tests an impossible scenario.